src/protocol.rs

```diff
@@ -38,73 +27,44 @@ use tracing::{error, info, instrument, trace};
 // Participant API //
 /////////////////////
 
+#[derive(Debug)]
+pub enum ProtocolType {
+    Keygen,
+    AuxInfo,
+    Presign,
+    Sign,
+    Broadcast,
+}
+
 /// Each participant has an inbox which can contain messages.
 #[derive(Debug)]
-pub struct Participant {
+pub struct Participant<P>
+where
+    P: PublicProtocolParticipant,
+{
     /// A unique identifier for this participant
     pub id: ParticipantIdentifier,
     /// A list of all other participant identifiers participating in the
     /// protocol
     pub other_participant_ids: Vec<ParticipantIdentifier>,
-    /// Local storage for this participant to store finalized auxinfo, keygen,
-    /// and presign values. This storage is not responsible for storing
-    /// round-specific material.
-    main_storage: Storage,
```

Comment:
:fireworks: ! Does this mean we can remove `Storage` entirely?

Reply:
I think so! I slapped an `#[allow(unused)]` on there at some point in this PR to get through an intermediary state, but I think we might be set to jettison it entirely. Will double-check.

Reply:
It's gone!!